dev/fattest.simplicity/src/componenttest/topology/utils/LDAPUtils.java

```diff
@@ -201,20 +201,39 @@ private static void initializeRemoteLdapServers() throws Exception {
         try {
             services = getLdapServices(2, CONSUL_LDAP_AD_SVT_SERVICE);
 
+            /**
```

Comment:
It seems like all this logic should be in `getLdapServices` so we don't have to replicate it.

If you check that method you will see we request X LDAP services from Consul, if we can't get X, we try X-1, etc until we either get at least 1 LDAP service.

I think we could do something like:

```
                List<ExternalTestService> services = new ArrayList<ExternalTestService>(ExternalTestService.getServices(requested, service));

                // NEW CODE STARTS HERE!!!!
                /*
                 * Remove services that are not available.
                 */
                Set<ExternalTestService> toRemove = new HashSet<ExternalTestService>();
                for (ExternalTestService service : services) {
                    if (!isLdapServerAvailable(service...) ) {
                       toRemove.add(service);
                    }
                }
                services.removeAll(toRemove);
                
                if (services.isEmpty()) {
                   // NO AVAILABLE LDAP SERVERS, FAIL BACK TO LOCAL????
                }
                // NEW CODE STOPS HERE!!!!
                
                /*
                 * Copy unique ...
                 */
```

Note: the isLdapServerAvailable may never return false. Looking at it, it appears to throw an Exception so that code above may need to change, or the method should perhaps be written better.

Reply:
Made updates -- `isLdapServerAvailable` usually throws an Exception. Since the method is called in several areas, I decided the risk wasn't worth the reward to rewrite it right now.